idx/memory/memory.go

```diff
@@ -269,6 +309,55 @@ func (m *MemoryIdx) GetPath(orgId int, path string) []idx.Archive {
 	return archives
 }
 
+func (m *MemoryIdx) IdsByTagQuery(orgId int, query tag.TagQuery) []string {
+	m.RLock()
+	defer m.RUnlock()
+
+	tree, ok := m.Tags[orgId]
+	if !ok {
+		return nil
+	}
+
+	return m.idsByTagQuery(tree, query)
+}
+
+func (m *MemoryIdx) idsByTagQuery(tags map[string]map[string][]string, query tag.TagQuery) []string {
```

Comment:
Is this function supposed to be returning the ids that match ANY expression or ALL expressions? Seems like it's ANY right now. I suppose this function is designed to be composable to support ALLs?

Reply:
@replay was confused, it should be ALL.  He's sorting that out now

Reply:
Fixed that with https://github.com/raintank/metrictank/pull/729/commits/ade96a417ecdff5bd35811e551521169a429d45e